src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorStatModule.java

```diff
@@ -78,41 +81,29 @@ public void onEquip(IToolStackView tool, ModifierEntry modifier, EquipmentChange
   public void onUnequip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
     if (condition.matches(tool, modifier)) {
       context.getTinkerData().ifPresent(data -> {
-        addStatIfArmor(tool, context, key, amount.compute(data.get(key2, 0)), allowBroken, heldTag);
-      });    }
+        addStatIfArmor(tool, context, statKey, amount.compute(data.get(key, 0f)), allowBroken, heldTag);
+      });
+    }
   }
 
   @Override
   public void addTooltip(IToolStackView tool, ModifierEntry modifier, @Nullable Player player, List<Component> tooltip, TooltipKey tooltipKey, TooltipFlag tooltipFlag) {
     if (condition.matches(tool, modifier) && (tool.hasTag(TinkerTags.Items.WORN_ARMOR) || heldTag != null && tool.hasTag(heldTag)) && (!tool.isBroken() || allowBroken)) {
-      float value = amount.compute(modifier.getEffectiveLevel());
-      if (value != 0) {
-        Component name = Component.translatable(Util.makeTranslationKey("armor_stat", key.getId()));
-        switch (tooltipStyle) {
-          case BOOST -> TooltipModifierHook.addFlatBoost(modifier.getModifier(), name, value, tooltip);
-          case PERCENT -> TooltipModifierHook.addPercentBoost(modifier.getModifier(), name, value, tooltip);
-        }
+      if (player != null) {
+        player.getCapability(TinkerDataCapability.CAPABILITY).ifPresent(data -> {
```

Comment:
You can use the helper here. `ArmorStatModule.getStat(...)`. Also, if the player is null use the modifier level. Should look something like:

```java
float level = player != null ? ArmorStatModule.getStat(player, levelKey) : modifier.getEffectieLevel();
```